app/com/arpnetworking/metrics/portal/scheduling/JobRefSerializer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.metrics.portal.scheduling;
+
+import java.util.Optional;
+
+/**
+ * A serializer that provides a way to map entity ids to  job refs and vice-versa.
+ * <p>
+ * This is used as somewhat of a crutch for the lack of dynamic props in Akka's
+ * classic actor clustering, and would be unnecessary if we later move to typed
+ * actors.
+ * <p>
+ * See <a href=https://stackoverflow.com/a/26524666>this SO post</a> for more details. This is Option A.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public interface JobRefSerializer {
```

Comment:
I'm not tied to the name, open to bikeshedding.

Reply:
I don't think we have an existing pattern for this in general. 

We use Jackson serde classes where needed in the case of JSON. Arguably we could use Jackson's framework here since it supports class to class conversions. I'm not a fan of this.

More generically, MAD has [Parser.java](https://github.com/ArpNetworking/metrics-aggregator-daemon/blob/master/src/main/java/com/arpnetworking/metrics/common/parsers/Parser.java) which fulfills half that contract.

AFAIK there's nothing general in Guava or the JDK.

I'd be up for seeing where this goes with: `com.arpnetworking.commons.serialization.Serializer` and `com.arpnetworking.commons.serialization.Deserializer`. Double parameterized (in & out types) with a single `serialize` and `deserialize` method each. Or more generically, a `Converter` class with a single method (serialization and deserialization are really the same process, just with different names and implementations -- if you assume a non-fixed target type).